### PR TITLE
modify SD dump to dump single partitions only 

### PR DESCRIPTION
**Pull request by mstormi**
Fixes: #1151  
(hopefully)
raw device partitions 1 (/boot) and 2 (/) only to avoid overwriting partition table or subsequent partitions

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

**Comment by ecdye** [comment]
Did you test and confirm it works for you?

File: functions/backup.bash [hunk 1]
```diff
@@ -353,7 +353,10 @@ mirror_SD() {
   local storageDir="${storagedir:-/storage}"
   local syncMount="${storageDir}/syncmount"
   local dirty="no"
-  
+  local dumpInfoText="For your information as the operator of this openHABian system:\\nA timed background job to run semiannually has just created a full raw device copy of your RPI's internal SD card.\\nOnly partitions to contain openHABian (/boot and / partitions 1 & 2) were copied."
```
Comment (ecdye) [inline functions/backup.bash:356]: Twice a year?

Comment (mstormi) [reply functions/backup.bash]: yep. We can change that later. Just killed an SD card so got more cautious, rsync also just once a week

**Comment by mstormi** [comment]
> 
> 
> Did you test and confirm it works for you?

still testing

**Comment by mstormi** [comment]
it doesn't work out.
you can merge the code but it doesn't boot from the alternative disk.
Then again, the previous code didn't either.
Trouble is the duplicated UUID when both cards are in (and they need to be else we cannot automatically keep the external one up to date). It's coincidence which device is booted from when both are in.

**Comment by ecdye** [comment]
Huh, odd...  Make sure you fix shellcheck errors before merging

**Comment by mstormi** [comment]
Found the issue, fstab also references PARTUUIDs not instead of device names so I had to replace those entries, too.
Right now busy testing.

**Comment by mstormi** [comment]
It's working fine now. Please review again and also test when your reader has arrived.

**Comment by ecdye** [comment]
> It's working fine now. Please review again and also test when your reader has arrived.

I actually have them here now and have started testing, I just have not had enough time to test this yet as I have been trying to finish up #1121

File: functions/backup.bash [hunk 2]
```diff
@@ -375,30 +369,56 @@ mirror_SD() {
...
+    #origPartUUID=$(blkid "${src}p2" | sed -n 's|^.*PARTUUID="\(\S\+\)".*|\1|p')
+    origPartUUID=$(blkid "${src}p2" | sed -n 's|^.*PARTUUID="\(\S\+\)".*|\1|p' | sed -e 's/-02//g')
+    if ! partUUID=$(yes | cond_redirect set-partuuid "${dest}2" random | awk '/^PARTUUID/ { print substr($7,1,length($7) - 3) }'); then echo "FAILED (set random PARTUUID)"; dirty="yes"; fi
+    if ! cond_redirect tune2fs "${dest}2" -U random; then echo "FAILED (set random UUID)"; dirty="yes"; fi
+    mount "${dest}1" "$syncMount"
+    sed -i "s|${origPartUUID}|${partUUID}|g" "${syncMount}"/cmdline.txt
+    umount "$syncMount"
+    mount "${dest}2" "$syncMount"
+    sed -i "s|${origPartUUID}|${partUUID}|g" "${syncMount}"/etc/fstab
+    sed -i 's|^What=.*|What=/dev/mmcblk0p3|g' "${syncMount}/etc/systemd/system/${storageDir}".mount
```
Comment (ecdye) [inline functions/backup.bash:397]: Will this get reset by syncs?

**Review by ecdye — APPROVED** [review]
LGTM

File: functions/backup.bash (hunk 2 above)
Comment (mstormi) [reply functions/backup.bash:397]: This code *is* what is being executed on every sync so it's effectively kept.
That's the storage mount which would point to the (no more existing) external SD after we had to exchange cards.

Comment (ecdye) [reply functions/backup.bash:396]: OK, I just misread the code.

**Pull request merged**